SwiftUIBasics/Views/SignUpView.swift

```diff
@@ -6,40 +6,191 @@
 //
 
 import SwiftUI
+import Combine
+
+class SignUpViewModel: ObservableObject {
+    // inputs
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var passwordConfirm: String = ""
+
+    // outputs
+    @Published var isValidUsernameLength: Bool = false
+    @Published var isValidPasswordLength: Bool = false
+    @Published var isValidPasswordUpperCase: Bool = false
+    @Published var isValidPasswordLowerCase: Bool = false
+    @Published var isValidPassWordOneSymbol: Bool = false
+    @Published var isValidPasswordOneNumber: Bool = false
+    @Published var isValidPasswordMatch: Bool = false
+    @Published var isValid: Bool = false
+    @Published var isValidPre: Bool = false
+
+    private var cancelableSet: Set<AnyCancellable> = []
+
+    init() {
+        $email
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[a-zA-Z0-9_]+([.][a-zA-Z0-9_]+)*@[a-zA-Z0-9_]+([.][a-zA-Z0-9_]+)*[.][a-zA-Z]{2,5}"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidUsernameLength, on: self)
+            .store(in: &cancelableSet)
+
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                return password.count >= 8
+            }
+            .assign(to: \.isValidPasswordLength, on: self)
+            .store(in: &cancelableSet)
+
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[A-Z]"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidPasswordUpperCase, on: self)
+            .store(in: &cancelableSet)
+        
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[a-z]"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidPasswordLowerCase, on: self)
+            .store(in: &cancelableSet)
+        
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[!\"#$%&'()*+,-./:;<=>?@\\[\\\\\\]^_`{|}~]+"
```

Comment:
```suggestion
                let pattern = "[\\W]+"
```
Non word regex works better